proto/aserto/directory/model/v3/model.proto

```diff
@@ -6,44 +6,114 @@ option go_package = "github.com/aserto-dev/go-directory/aserto/directory/model/v
+    ON_DELETE_DELETE                                              = 0x2;
+    // delete all types before creating the new ones
+    ON_DELETE_NUKE                                                = 0x3;
+}
+
+message MetadataReq {
+    Metadata metadata                                             = 1       [(google.api.field_behavior) = REQUIRED];
+    // operation code indicating how to handle types deletion
+    OnDelete op_code                                              = 2       [(google.api.field_behavior) = REQUIRED];
+}
+
+message GetManifestRequest {
+    Metadata request                                              = 1       [(google.api.field_behavior) = REQUIRED];
 }
 
 message GetManifestResponse {
-    // manifest instance
-    aserto.directory.common.v3.Manifest result                  = 1;
+    oneof msg {
+        Metadata metadata                                         = 1;     // Manifest metadata  
+        bytes   body                                              = 2;     // Manifest content
+    }
 }
 
 message SetManifestRequest {
     // manifest instance
-    aserto.directory.common.v3.Manifest manifest                = 1     [(google.api.field_behavior) = REQUIRED];
+    oneof msg {
+        MetadataReq metadata                                      = 1;     // Manifest metadata 
+        bytes   body                                              = 2;     // Manifest content
+    }
 }
 
 message SetManifestResponse {
-    // manifest instance
-    aserto.directory.common.v3.Manifest result                  = 1;
+    google.protobuf.Empty result                                  = 1;    // empty result
 }
 
 message DeleteManifestRequest {
-    // manifest name (unique, lc-string)
-    string name                                                 = 1     [(google.api.field_behavior) = REQUIRED];
-    // manifest version (semver notation)
-    string version                                              = 2     [(google.api.field_behavior) = REQUIRED];
+    Metadata request                                              = 1      [(google.api.field_behavior) = REQUIRED];
 }
 
 message DeleteManifestResponse {
-    // empty result
-    google.protobuf.Empty result                                = 1;
+    google.protobuf.Empty result                                  = 1;    // empty result
+}
+
+message ListManifestsRequest {
+    aserto.directory.common.v3.PaginationRequest page_request     = 1;    // page request
+}
+
+message ListManifestsResponse {
+    repeated Metadata manifests                                   = 1;    // manifests metadata
+    aserto.directory.common.v3.PaginationResponse next_page       = 2;    // next page token
+}
+
+
+
+
+
+message ObjectType {
+    // object type name (unique, lc-string)
+    string name                                                   = 1      [(google.api.field_behavior) = REQUIRED];
+    repeated RelationTypes relation_types                         = 2;
```

Comment:
Would it be more convenient to use if `relation_types` was a `map<string, RelationType>` with the relation type name as the key?
It would make it easy to look up a relation type by its name and object type.